packages/v4-sdk/src/poolKey.ts

```diff
@@ -0,0 +1,34 @@
+import { Address, encodeAbiParameters, keccak256, parseAbiParameters, zeroAddress } from 'viem'
+import { Bytes32 } from './types'
+import { BinPoolParameter, CLPoolParameter, encodePoolParameters } from './utils/encodePoolParameters'
+
+export type PoolType = 'CL' | 'Bin'
+
+/**
+ * PoolKey is a unique identifier for a pool
+ *
+ * @see PoolKey
+ */
+export type PoolKey<T extends PoolType = never> = {
+  currency0: Address
+  currency1: Address
+  hooks?: Address
+  poolManger: Address
+  fee: number
+  parameters: T extends 'CL' ? CLPoolParameter : T extends 'Bin' ? BinPoolParameter : CLPoolParameter | BinPoolParameter
+}
+
+export const toId = ({ currency0, currency1, hooks = zeroAddress, poolManger, fee, parameters }: PoolKey): Bytes32 => {
```

Comment:
Can we rename to `getPoolId`? `toId` makes more sense if it has more context like `poolKey.toId`, while it's a pure function here